### ISSUE TITLE
GitHub Actions error: `Error: fetch failed`

### ISSUE THREAD

**Issue by gaggle** [pkgxdev/setup#170]
Got this [all over my CI](https://github.com/gaggle/deliverymetrics/actions/runs/5672658868/attempts/1):
![Screenshot 2023-07-26 at 21 00 26](https://github.com/teaxyz/setup/assets/2316447/68a6bc03-acd5-4265-9c74-b6444c4f8990)

It [ran successfully when I retried the job](https://github.com/gaggle/deliverymetrics/actions/runs/5672658868). Is there retry logic to consider adding to increase fetching resilience?

(and thanks for making this amazing project available)

**Comment by jhheider** [comment]
This seems like a good idea. GitHub actions seems to suffer intermittent failures of connectivity even during runs. pantry's closed issues are a testament to this.

**Comment by mxcl** [comment]
seems like a good plan yeah. three reties perhaps with a small sleep?

**Comment by jhheider** [comment]
sleep 1; sleep 3; sleep 5? probably not worth including any kind of sophisticated expo backoff tech.

**Comment by mxcl** [comment]
here we had a fetch hang for 2 hours:

<img width="872" alt="image" src="https://github.com/teaxyz/setup/assets/58962/0dfd5a42-f294-417a-993d-8c10a5441298">

**Comment by jhheider** [comment]
Yep. Sometimes an action step will hang for 6 hours then time out. I get github status texts most days of the week that some aspect of their infrastructure is degraded. :/

**Comment by mxcl** [comment]
oh so not likely because our code sucks?

**Comment by jhheider** [comment]
Anything is possible, of course.

**Comment by mxcl** [comment]
`v3` is a rewrite that shouldn’t have these issues. Provided rust is more reliable in `pkgx` itself that is.
